recipes/godot-cpp/all/conanfile.py

```diff
@@ -12,131 +10,43 @@ class GodotCppConan(ConanFile):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
 
-        compiler = str(self.settings.compiler)
-        if compiler not in minimal_version:
-            self.output.warn(
-                "{} recipe lacks information about the {} compiler standard version support".format(self.name, compiler))
-            self.output.warn(
-                "{} requires a compiler that supports at least C++{}".format(self.name, minimal_cpp_standard))
-            return
+    def export_sources(self):
+        export_conandata_patches(self)
 
-        version = tools.Version(self.settings.compiler.version)
-        if version < minimal_version[compiler]:
-            if compiler in ["apple-clang", "clang"]:
-                raise ConanInvalidConfiguration(
-                    "{} requires a clang version that supports the '-Og' flag".format(self.name))
-            raise ConanInvalidConfiguration(
-                "{} requires a compiler that supports at least C++{}".format(self.name, minimal_cpp_standard))
-
-    def build(self):
-        self.run("python  --version")
-        if self.settings.os == "Macos":
-            self.run("which python")
-        self.run("scons  --version")
-        self.run(
-            " ".join([
-                "scons",
-                "-C{}".format(self._source_subfolder),
-                "-j{}".format(tools.cpu_count()),
-                "generate_bindings=yes",
-                "use_custom_api_file=yes",
-                "bits={}".format(self._bits),
-                "custom_api_file={}".format(self._custom_api_file),
-                "headers_dir={}".format(self._headers_dir),
-                "platform={}".format(self._platform),
-                "target={}".format(self._target),
-                "use_llvm={}".format(self._use_llvm),
-                "use_mingw={}".format(self._use_mingw),
-            ])
-        )
+    def layout(self):
+        cmake_layout(self)
 
     def package(self):
-        self.copy("LICENSE*", dst="licenses", src=self._source_subfolder)
-        self.copy("*.hpp", dst="include/godot-cpp", src=os.path.join(self._source_subfolder, "include"))
-        self.copy("*.a", dst="lib", src=os.path.join(self._source_subfolder, "bin"))
-        self.copy("*.lib", dst="lib", src=os.path.join(self._source_subfolder, "bin"))
+        cmake = CMake(self)
+        cmake.install()
 
     def package_info(self):
-        if self.settings.os == "Windows" and self.settings.compiler == "Visual Studio":
-            self.cpp_info.libs = ["lib{}".format(self._libname)]
+        platform = {
+            "Android": "android",
+            "Linux": "linux",
+            "Macos": "darwin",
+            "Windows": "windows",
+        }[self.settings.get_safe("os")]
+        build_type = self.settings.get_safe("build_type").lower()
+
+        if self.settings.get_safe("os") == "Android":
+            self.cpp_info.libs= [f"godot-cpp.{platform}.{build_type}"]
```

Comment:
```suggestion
        }[self.settings.os]
        build_type = self.settings.build_type.lower()

        if self.settings.os == "Android":
            self.cpp_info.libs= [f"godot-cpp.{platform}.{build_type}"]
```

Reply:
`build_type`, `os` and `arch` are standard settings that are guaranteed to be available.

Reply:
adjusted by using `str(...)` where a string-like or hashable object is required.